docs/faq.md

```diff
@@ -1,21 +1,321 @@
+
+## How does scaling work? Should this be initiated by a customer?
+
+Fn runs more instances of the function based on requests.  It is automatic.
+
+## When one host is overloaded, will this service automatically relocate the function to another host?
+
+When running on Kubernetes (K8s) workloads are spread across available hosts.
+
+Functions currently do not relocate. Please see our [FnLB project inside Fn Project](https://github.com/fnproject/fn/tree/master/fnlb) for how to route requests to available hosts. K8s can also help with this.
+
+# Java
+
+## Any Java runtime dependencies? Can we use concurrent libraries or any third party?
+
+We currently support Maven which allows you to specify your dependencies. Therefore, you can use concurrency and third party libraries.
+
+## What versions of Java are supported (JDK 8, JDK 9)?
+
+We default to Java 9 now but if you want Java 8, the runtime name is `java8` (when you init a function or in your `func.yaml`).
+
+# Persistence
+
+## Does your runtime provide any persistent store to save the function state information? 
+
+State management is not part of Fn but you can use any storage service or database to store data.
+
+## Can we build a state function which will update itself based on the previous computation? Will it persist across function restarts?
+
+The problem you face is the lack of a guarantee of which instance of a hot function is called.  Standard practice is to externalize state.  The State Service is essentially Oracle Coherence which offers excellent performance as you may know.
+
+Also check out Fn Flow if you’re writing Java (currently the primary supported language).
+
+## Do you provide any UI/tool to track our Fn status?
+
+Fn has a dashboard that can be found at <https://github.com/fnproject/ui>. Fn Flow also has an experimental dashboard at <https://github.com/fnproject/flowui>. 
+
+# Fn Java Function Development Kit (FDK)
+
+## What is the Fn Java FDK?
+
+The Fn Java FDK helps you build, test and run Java functions on the Fn platform.
+
+The FDK is comprised of:
+
+* a Java library (known as the `api`)
+* a runtime Docker image (known as the `runtime`)
+* a JUnit rule
+* a build-time Docker image for repeatable builds
+
+## Is the FDK required in order to run Java on Fn?
+
+No. You can still write Java functions on Fn without using the FDK. However using the FDK will make several things easier for you:
+
+1. A curated base image for Java 8 and Java 9 means that you don't have to build and maintain your own image. These images contain optimizations for quick JVM startup times.
+2. Accessing configuration from Fn is easy through FDK APIs.
+3. Input and output type coercion reduces the amount of serialization and formatting boilerplate that you have to write.
+4. A JUnit rule provides a realistic test harness for you to test your function in isolation.
+
+# FN Flow
```

Comment:
I'd keep the "What is Fn Flow" question below with it's answer, then move the rest of this section to the appropriate repos (either flow/completer repo or fdk-java repo).

Reply:
Copied the first question to General and removed the rest.